internal/auth/authenticator.go

```diff
@@ -0,0 +1,95 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package auth
```

Comment:
Only the interface, or the whole package? In any case, I would prefer to keep the config package clean from the implementation, as seems to be the case today.

Reply:
that's why I am suggesting an internal package inside the configauth which will not be exported. But this way we have localization, the current internal/auth is only used by `configauth`

Reply:
Except that the gRPC and HTTP receivers will use a default implementation for the interceptors/handlers, also located in this package.

Reply:
Ok, let's move forward with this (I don't believe that we cannot just have it in the internal here). But let's see.

Reply:
I'll collect the non-blocking comments in an issue, so that I can solve them in a follow-up PR. To be honest, my biggest contention right now in changing this PR is in rebasing the other two PRs again.

Reply:
https://github.com/open-telemetry/opentelemetry-collector/issues/1824